File: week-3/assign-cookies/assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        
        c_index = 0
        cookie_index = 0
        while c_index < len(g) and cookie_index < len(s):
            if s[cookie_index] >= g[c_index]:
                c_index += 1
            cookie_index += 1
        
        return c_index

File: week-3/assign-cookies/assign-cookies.js
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function(g, s) {
    g = g.sort((a,b)=>a-b);
    s = s.sort((a,b)=>a-b);
    
    let c_index = 0, cookie_index = 0;
    while (c_index < g.length && cookie_index < s.length) {
        if (s[cookie_index] >= g[c_index]) ++c_index;
        ++cookie_index;
    }
    
    return c_index;
};
